frontend/src/pages/integrations/railway/create.tsx

```diff
@@ -121,106 +144,138 @@ export default function RailwayCreateIntegrationPage() {
   };
 
   return workspace &&
-    selectedSourceEnvironment &&
+    getValues("selectedSourceEnvironment") &&
     integrationAuthApps &&
     targetEnvironments &&
     filteredServices ? (
     <div className="flex h-full w-full items-center justify-center">
       <Card className="max-w-md rounded-md p-8">
         <CardTitle className="text-center">Railway Integration</CardTitle>
-        <FormControl label="Project Environment" className="mt-4">
-          <Select
-            value={selectedSourceEnvironment}
-            onValueChange={(val) => setSelectedSourceEnvironment(val)}
-            className="w-full border border-mineshaft-500"
-          >
-            {workspace?.environments.map((sourceEnvironment) => (
-              <SelectItem
-                value={sourceEnvironment.slug}
-                key={`source-environment-${sourceEnvironment.slug}`}
-              >
-                {sourceEnvironment.name}
-              </SelectItem>
-            ))}
-          </Select>
-        </FormControl>
-        <FormControl label="Secrets Path">
-          <Input
-            value={secretPath}
-            onChange={(evt) => setSecretPath(evt.target.value)}
-            placeholder="Provide a path, default is /"
+        <form onSubmit={handleSubmit(handleButtonClick)}>
+          <Controller
+            name="selectedSourceEnvironment"
+            control={control}
+            render={({ field }) => {
+              return (
+                <FormControl label="Project Environment" className="mt-4">
```

Comment:
Kindly include error message from yup to the formControl like [this](https://github.com/Infisical/infisical/blob/main/frontend/src/views/Settings/OrgSettingsPage/components/OrgNameChangeSection/OrgNameChangeSection.tsx#L64C6-L64C6)